partiql-plan/src/main/java/org/partiql/plan/rel/RelSort.java

```diff
@@ -0,0 +1,87 @@
+package org.partiql.plan.rel;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.plan.Collation;
+import org.partiql.plan.Operator;
+import org.partiql.plan.OperatorVisitor;
+
+import java.util.List;
+
+/**
+ * Logical sort abstract base class.
+ */
+public abstract class RelSort extends RelBase {
+
+    /**
+     * @return new {@link RelSort} instance
+     */
+    @NotNull
+    public static RelSort create(@NotNull Rel input, @NotNull List<Collation> collations) {
+        return new Impl(input, collations);
+    }
+
+    @NotNull
+    public abstract Rel getInput();
+
+    @NotNull
+    public abstract List<Collation> getCollations();
+
+    @NotNull
+    @Override
+    protected final RelType type() {
+        return getInput().getType();
+    }
+
+    @NotNull
+    @Override
+    protected final List<Operator> operands() {
+        Rel c0 = getInput();
+        return List.of(c0);
```

Comment:
Need to pass the collations

Reply:
Collations are not operators, so they're not operands. For RelSort, the only operand is the "input" and the "collations" list is an argument.

Reply:
Yeah, this comment has the same reasoning as https://github.com/partiql/partiql-lang-kotlin/pull/1665#discussion_r1870363550, but this one could potentially be disregarded since I see that the Rex in the collation will be replaced by an `int`.